cmake/compiler/gcc/target.cmake

```diff
@@ -110,6 +110,11 @@ assert_exists(LIBGCC_DIR)
 LIST(APPEND LIB_INCLUDE_DIR "-L\"${LIBGCC_DIR}\"")
 LIST(APPEND TOOLCHAIN_LIBS gcc)
 
+# We should use built-in CMAKE_C_STANDARD_LIBS (and friends) instead of re-inventing the wheel.
+# This setting should ideally be based upon toolchain coupled with information related to c++ config.
+# Todo: deprecate TOOLCHAIN_LIBS, it appears it tries to achieve what CMAKE_C_STANDARD_LIBRARIES provides.
+set(CMAKE_C_STANDARD_LIBRARIES "-lc -lgcc" CACHE INTERNAL "")
```

Comment:
This will require that all C libraries use the same name, which isn't always true (iirc newlib nano uses -lc_nano?) Not a big deal, but just something to keep in mind.

Reply:
Also, I assume cmake places all of these libraries at the end of the link command? That's not going to work for us; we need to include libc (and probably libgcc) in the loop.

Reply:
> This will require that all C libraries use the same name, which isn't always true

Correct, but that goes for all our solutions.
But for the newlib nano, then specifying `--specs=nano.specs` is sufficient.

>  I assume cmake places all of these libraries at the end of the link command?

Correct.

> That's not going to work for us; we need to include libc (and probably libgcc) in the loop.

Correct, it's not solving, and was not intended to solve the circular dependency.
However, it is a good practice to get the standard libs linked at the end, as we cannot prevent some Zephyr module code or user app using vanilla CMake but not linking against kernel lib.
However, to also ensure such cases to always work, then we might want to expand that line a bit.

It was more one of those small issues I discovered along the way.

Reply:
Any random Zephyr module is almost certainly going to use the C library API, which means it implicitly links against the kernel lib -- the C library depends upon it. So, some way of getting those libraries inserted into the link command ahead of the dependency loop including the C library and the kernel is necessary. If there's a way to make this work with the regular cmake functions, that would be great. Right now, depending on `zephyr_link_libraries` changes at least gets things working.